pydantic_ai_slim/pydantic_ai/_agent_graph.py

```diff
@@ -143,6 +143,24 @@ def is_agent_node(
     return isinstance(node, AgentNode)
 
 
+def _create_thinking_retry_request(parts: list[_messages.ModelResponsePart]) -> _messages.ModelRequest | None:
+    """Handle thinking-only responses (responses that contain only ThinkingPart instances).
+
+    This can happen with models that support thinking mode when they don't provide
+    actionable output alongside their thinking content.
+    """
+    thinking_parts = [p for p in parts if isinstance(p, _messages.ThinkingPart)]
+    if thinking_parts:
```

Comment:
I think we can drop this check and always try to get the model to try again instead of raising a hard error.